protocols/raft/src/main/java/io/atomix/protocols/raft/metrics/RaftRoleMetrics.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.protocols.raft.metrics;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.Histogram;
+
+public class RaftRoleMetrics extends RaftMetrics {
+
+  private static final Gauge ROLE =
+      Gauge.build()
+          .namespace("atomix")
+          .name("role")
+          .help("Shows current role")
+          .labelNames("partition")
+          .register();
+
+  private static final Counter HEARTBEAT_MISS =
+      Counter.build()
+          .namespace("atomix")
+          .name("heartbeat_miss_count")
+          .help("Count of missing heartbeats")
+          .labelNames("last", "partition")
+          .register();
+
+  private static final Histogram HEARTBEAT_TIME =
+      Histogram.build()
+          .namespace("atomix")
+          .name("heartbeat_time_in_s")
+          .help("Time between heartbeats")
+          .labelNames("partition")
+          .register();
+
+  public RaftRoleMetrics(String partitionName) {
+    super(partitionName);
+  }
+
+  public void becomingFollower() {
+    ROLE.labels(partition).set(1);
+  }
+
+  public void becomingCandidate() {
+    ROLE.labels(partition).set(2);
+  }
+
+  public void becomingLeader() {
+    ROLE.labels(partition).set(3);
+  }
+
+  public void countHeartbeatMiss(long lastHeartbeat) {
+    HEARTBEAT_MISS.labels(Long.toString(lastHeartbeat), partition).inc();
```

Comment:
Could this result in us having a very large number of label values? 

From their [documentation](https://prometheus.io/docs/practices/naming/):

> CAUTION: Remember that every unique combination of key-value label pairs represents a new time series, which can dramatically increase the amount of data stored. Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.

Reply:
ah forgot to remove that thanks